services/motion/explore/explore.go

```diff
@@ -332,7 +343,16 @@ func (ms *explore) checkForObstacles(
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
-			currentPose := spatialmath.NewZeroPose()
+
+			currentInputs, err := kb.CurrentInputs(ctx)
+			if err != nil {
+				ms.logger.Debugf("issue occurred getting current inputs from kinematic base: %v", err)
```

Comment:
why not return?

Reply:
Great question. I was debating whether returning or sending a message on the obstacle channel would be appropriate for both the error case involving `CurrentInputs` and `generateTransientWorldState`. 

My thoughts are that since we are looping here there is a chance we may have recovered by the next iteration and can continue towards our initial point instead of breaking out a defining a new one. However, such a recovery is not guaranteed. If we don't recover there is a change that we will continue to move straight until the plan execution is complete. 

If this issue is uncommon and unrecoverable, I think you are right we need to tell the main program to try a new point (reconstructing the kinematic base in the process), not solely through a return but also through a message to the `obstacleResponseChan` channel least we get stuck waiting for a message from execute plan similarly to our current setup. 

If this issue is common but is recovered from on the next loop, then it may be better to just log and ignore these issues as they are more than likely to get resolved by the next iteration. 

I'm not sure if I currently have enough data to know which is scenario is more likely, although I'm happy to look into the matter.

Reply:
Done